### ISSUE TITLE
Inefficient patch padding

### ISSUE THREAD

**Issue by qgallouedec** [huggingface/jat#57]
I share with you a difficulty, I would like your advice.

Not all datasets have the same modalities. In particular, in some cases, the observation is an image, and in others not.
To mix all these datasets in one, we fill the missing modalities with a padding value. To simplify, we go from

```
[
    {"input_ids": [123, 456, ...]}  # these are tokens
    {"patches": [P1, P2, ...]}    # these are patches of shape (4, 16, 16)
]
```

to

```
[
    {"input_ids": [123, 456, ...],
     "patches": [P0, P0, ...], # patch pad
     "input_types": [0, 0, ...]} # to specify that input are tokens
    {"input_ids": [0, 0, ...], # token pad
     "patches": [P1, P2, ...],
     "input_types": [1, 1, ...]} # to specify that input are patches
]
```

The problem is that padding on patches is very expensive in memory, because each pad costs a patch of 4 x 16 x 16 uint8
The way out of this (and this is what the processor is currently doing) is to use a single object for all the pads. It works very well.

But here is where it gets tricky:
The dataset used by `Trainer` must be an instance of `datasets.Dataset`. However, when we instantiate this class with our dataset (`dataset = Dataset.from_dict(dataset)`), a new object is created for each value in the dataset, in other words, the trick explained before is cancelled. Result: very long process, and very costly in memory, only to write lots of zeros that will be masked in the end.

Here is a code to reproduce this.

```python
from datasets import Dataset
from transformers import Trainer

from gia.config import Arguments
from gia.datasets import collate_fn, load_gia_dataset
from gia.model import GiaModel
from gia.processing import GiaProcessor

# Initialize the processor and model
args = Arguments(task_names=["mujoco-ant"], output_dir="./")
processor = GiaProcessor(args)
model = GiaModel(args)

# Load the dataset
train_dataset = load_gia_dataset(task_names=args.task_names, split="train")
train_dataset = processor(**train_dataset)
train_dataset = Dataset.from_dict(train_dataset) # <- This line
trainer = Trainer(model, args, data_collator=collate_fn, train_dataset=train_dataset)
trainer.train()
```
